cmd/finch-daemon/main.go

```diff
@@ -69,6 +70,31 @@ func runAdapter(cmd *cobra.Command, _ []string) error {
 	return run(options)
 }
 
+func getListener(options *DaemonOptions) (net.Listener, error) {
+	var listener net.Listener
+	var err error
+
+	if options.socketAddr == "fd://" {
+		listeners, err := activation.Listeners()
+		if err != nil {
+			return nil, fmt.Errorf("cannot retrieve listeners: %w", err)
+		}
+		if len(listeners) != 1 {
+			return nil, fmt.Errorf("unexpected number of socket activations (%d != 1)", len(listeners))
+		}
+		listener = listeners[0]
+	} else {
+		listener, err = net.Listen("unix", options.socketAddr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to listen on %s: %w", options.socketAddr, err)
+		}
+		if err := os.Chown(options.socketAddr, options.socketOwner, options.socketOwner); err != nil {
```

Comment:
Does it make sense to set the socketOwner in the `fd://` case? Can't `--socket-owner` still be passed in the command line arguments in the systemd service file? The correct way is probably to use `SocketUser` and `SocketGroup` as you have in your example in the docs, but should we support the CLI option as well, or at least call out that they don't work together?

Reply:
i have added a check for it.